torch/cuda/amp/amp_scaler.py

```diff
@@ -0,0 +1,332 @@
+import torch
+from collections import defaultdict
+from torch._six import container_abcs
+
+
+class _PerDeviceHelper(object):
+    """
+    Helper class to serve copies of a master tensor that may be needed on multiple devices.
```

Comment:
Docstring should be clarified. Forget the idea of one tensor being a "master" tensor (although not the worst use of jargon). Maybe something like: "Helper class that replicates a CUDA tensor to other devices as needed." 

Do you want to check that the origin tensor given is on a cuda device?